### ISSUE TITLE
Unnecessary paddings in full-screen nvim

### ISSUE THREAD

**Issue by sergeyzwezdin** [warpdotdev/Warp#5563]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

When I set uniform padding to 0 and open nvim in full-screen there are still some paddings at the bottom and right (check the screenshots). Ideally, I would get rid of that.

### To reproduce

1. Set uniform padding to `0`
2. Open `nvim`
3. Go to full screen

### Expected behavior

No paddings if uniform padding settings set to `0`

### Screenshots

![2024-12-12 at 07 25 14@2x](https://github.com/user-attachments/assets/efc22403-b5b4-42ed-b16a-2d8b43728333)
![2024-12-12 at 07 30 15@2x](https://github.com/user-attachments/assets/ccb67fe4-a8e9-4e8c-ae99-80ccf8e33d62)

### Operating system

MacOS

### Operating system and version

15.1.1

### Shell Version

zsh 5.9 (arm-apple-darwin22.1.0)

### Current Warp version

v0.2024.12.10.15.55.stable_02

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Additional context

There is already #2373 where this bug initially mentioned, but @szgupta also mentioned `Please feel free to open up a new issue if you're running into any problems`, so opening it.

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hey @sergeyzwezdin, thanks for submitting this. This behavior is similar across many terminals, but not all. That being said, there is always room for improvement and getting back the most usable space possible for alt-screens.

To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by szgupta** [comment]
@sergeyzwezdin thanks for following up on this! Just to add on to what @dannyneira said: Warp allows you to scale your terminal size (and in turn, your window) by fractions of a cell width / height. So when your terminal size is not perfectly aligned to a cell width / height, the extra space appears as padding. Other terminals that support this kind of fractional scaling also have a similar effect. 

Other terminals (e.g. iTerm) snap the terminal size to perfectly align with full cell widths / heights, which avoids the issue entirely, but at the expense of not supporting fractional window scaling.

I'd love to hear if you have any suggestions / alternatives on how we can improve the UX here 😄

**Comment by dannyneira** [comment]
@sergeyzwezdin Fractional scaling in action

https://github.com/user-attachments/assets/f5324943-d5c5-49c4-b79d-4a7ed3794ec0

**Comment by slieschke** [comment]
Thanks for the explanation of where the bottom and right paddings come from @szgupta. That makes perfect sense. In light of that the new uniform padding option sets a somewhat misleading expectation; with it set to zero it's actually "zero top and left padding, and as close to zero bottom and right padding that can be done dependent on your window size". Perhaps your explanation could be incorporated to <https://docs.warp.dev/features/full-screen-apps#padding> that is linked from the settings UI?

My ideal solution would be an additional option to allow centring alt screen apps after the current uniform padding is applied, so the spacing around them is consistent on all sides. I totally understand however if adding another option isn't desirable; I could imagine it could be tricky to communicate effectively in the settings UI.

**Comment by szgupta** [comment]
@slieschke thanks for the feedback! We'll add a note to that page.

> My ideal solution would be an additional option to allow centring alt screen apps after the current uniform padding is applied, so the spacing around them is consistent on all sides.

We actually tried the centering approach first (as a default) to alleviate the effect of the padding caused by fractional scaling. But we also saw feedback like [this](https://x.com/willmcgugan/status/1862540652384284994) that made it unideal. I'm curious if you think there's a better approach!

**Comment by slieschke** [comment]
> But we also saw feedback like [this](https://x.com/willmcgugan/status/1862540652384284994) that made it unideal.

Arguably that's an acceptable tradeoff because resizing the terminal is an edge case compared to actually doing things in the terminal. But understandably people will see that behaviour and react unfavourably not understanding the reasoning behind it, in which case the extra (static) bottom/right padding is probably better because people are less likely to notice that compared the the (dynamic) resizing wobble.

> I'm curious if you think there's a better approach!

The resizing wobble could perhaps be minimised by removing the centring on the start of the resize, and adding it back after some period of no resizing. But that's getting into the realm of diminishing returns for extra code complexity; and it possibly brings in some other unanticipated undesirable behaviour as I've learnt here! 🤣️

**Comment by dannyneira** [comment]
Thanks again for your feedback @slieschke. I've added a note in our docs to help explain the padding in the cases of fractional scaling. https://docs.warp.dev/features/full-screen-apps#padding

**Comment by sergeyzwezdin** [comment]
> Fractional scaling in action

@dannyneira Yeah, that makes sense. Thank you for your explanation. Taking in mind that fact, I'm not annoyed anymore when see this padding :D

**Comment by dannyneira** [comment]
@sergeyzwezdin great! 🥳 I'll close since this was just a misunderstanding and is resolved with clarification in the docs. Please let us know if there are any issues with the padding or fractional scaling features.
